src/components/Input/AmountInput.tsx

```diff
@@ -39,16 +35,35 @@ export const AmountInput: React.FC<IProps> = ({
 
   const [open, setOpen] = useState(false)
 
-  const allowOnlyDigits = (e: inputString) => {
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const allowOnlyDigitsAndTrimUnnecessaryZeroes: React.ChangeEventHandler<HTMLInputElement> = (e) => {
     const regex = /^\d*\.?\d*$/
     if (e.target.value === '' || regex.test(e.target.value)) {
-      setValue(e.target.value)
+      const startValue = e.target.value
+      let parsed = e.target.value
+      const zerosRegex = /^0+\d+\.?\d*$/
```

Comment:
I am not sure if this is the correct input value:
![Screenshot from 2021-07-23 22-37-26](https://user-images.githubusercontent.com/43304803/126839011-e9119f19-cb82-4b45-a25d-e38db450975b.png)

reproduction:
- input value `[0.111]`
- adding zero to start `[0.111]` -> `[.111]`